Update README.md (#7)
larkox/mattermost-plugin-badges: README.md

# Badges plugin
Let your users show appreciation by granting badges.

## Install
Get the latest release from [GitHub](https://github.com/larkox/mattermost-plugin-badges/releases) and [install it manually](https://developers.mattermost.com/integrate/plugins/server/hello-world/#installing-the-plugin) on your server.

## Configuration

![Screenshot from 2022-03-16 11-02-13](https://user-images.githubusercontent.com/1933730/158565396-9d637c4c-6772-449f-81cb-2b73f8f6670e.png)

> - Badges admin: Every sysadmin is considered a badges admin. Apart of that, you can assign this role to one person by adding their username here. Only one person is permitted.

Author: Only one person as badge admin (plus all the sysadmins in the system). It can be improved, but it is the behaviour right now :sweat_smile:

## Usage
### Creating a type

> In order to create a type, you must be a badges admin. Types are used to organize badges and assign permissions.

Reviewer: ```suggestion
Badge admins can create different types of badges, and each type of badge can have its own permissions. You must be a badge admin to create a badge type.
```

Run the slash command `/badges create type` to open the creation dialog.

![Screenshot from 2022-03-16 11-14-31](https://user-images.githubusercontent.com/1933730/158567578-1241cc93-6964-4dc7-a56b-a5b3729229b7.png)

- Name: The name of the type. Will be shown in the badges description.
- Everyone can create badge: If you mark this checkbox, every user in your Mattermost instance can create badges of this type.
- Can create allowlist: This list contains the usernames (comma separated) of all the people allowed to create badges of this type.
- Everyone can grant badge: If you mark this checkbox, every user in your Mattermost instance can grant any badge of this type.
- Can grant allowlist: this list contains the usernames (comma separated) of all the people allowed to grant badges of this type.

### Permissions details

> Badge admins can always create types, create badges for any type, and grant badges from any type, no matter the permissions in place.

Reviewer: ```suggestion
Badge admins can always create types, create badges for any type, and grant badges from any type, regardless of the permissions in place for a given badge type.
```

> Badge creator can always grant the badge they created.

Reviewer: ```suggestion
A badge creator can always grant the badge they created.
```

- Everyone can create badges, but can only grant the badges they have created: ECC true, ECG false, CC empty, CG empty
- Everyone can create and grant any badge: ECC true, ECG true, CC empty, CG empty

### Creating a badge
In order to create a badge, you must decide the type of that badge, and have permissions to create a badge of that type.
Run the slash command `/badges create badge` to open the creation dialog.

![Screenshot from 2022-03-16 11-37-32](https://user-images.githubusercontent.com/1933730/158571687-4983f7e4-1cf9-4fa1-a3f1-6f80918e28e3.png)

- Name: Name of the badge.
- Description: Description of the badge.
- Image: Only emojis are allowed. You must input the emoji name as you would do to add it to a message (e.g. `:+1:` or `:smile:`). Custom emojis are also allowed.
- Type: The type of the badge. This list will show only types you have permissions to create.
- Multiple: Whether this badge can be granted more than once to the same person.

### Details about Multiple

> All badges can be assigned to any number of persons. What the Multiple setting controls is whether this badge can be granted more than once to the same person. For example, a "Thank you" badge should be granteable many times (many people can be thankful to you in more than one ocassion). This badge should have the Multiple checkbox marked. But a "First year in the company" badge should be granted just once (it makes no sense to have 2 first year in the company). That badge should have the Multiple checkbox unmarked.

Reviewer: ```suggestion
All badges can be assigned to any number of people. What the **Multiple** setting controls is whether this badge can be granted more than once to the same person. For example, a "Thank you" badge should be grantable many times (many people can be thankful to you on more than one occasion), and therefore, a Thank You badge should have the **Multiple** option selected. However, a "First year in the company" badge should be granted only once since a user won't celebrate this milestone multiple times at the same company. This type of badge should have the **Multiple** option unselected.
```

### Granting a badge
In order to grant a badge, you must have permissions to grant it.
There are two ways to open the grant dialog. One of them is using the `/badges grant` command. Another one is clicking on the "Grant badge" link available in the Profile Popover visible when you click on someones username.

![Screenshot from 2022-03-16 11-47-14](https://user-images.githubusercontent.com/1933730/158573673-723e77a2-6d58-4aa5-8a89-6adcbce50e13.png)

The dialog looks like this:

![Screenshot from 2022-03-16 11-51-05](https://user-images.githubusercontent.com/1933730/158573834-70ea72b0-4a03-4b09-a694-751c0ca1ba04.png)

- User: The user you want to grant the badge to (may be prepopulated if you clicked the grant button from the profile popover, or added the username in the command).
- Badge: The badge you want to grant (may be prepopulated if you added the badge id in the command).
- Reason: An optional reason why you are awarding this badge. (Specially useful for badges like "Thank you")

> - Notify on this channel: If you mark this, a message from the badges bot will be posted in the current channel, letting all know that you granted this badge to that person.

Reviewer: ```suggestion
- Notify on this channel: If you select this option, a message from the badges bot will be posted in the current channel, letting everyone in that channel know that you granted this badge to that person.
```

> When a badge is granted, three things may happen:

Reviewer: Given that the first bullet point below WILL always happen, consider rewording the introductory sentence as: 
"When a badge is granted, the user who received the badge will receive a DM from the badges bot letting them know that they have been awarded a badge."

Then, you can introduce the 2 potential outcomes as: "In addition, the following may happen:"

- The user that received the badge will always receive a DM from the badges bot letting them know they have been awarded a badge.
- If "Notify on this channel" was marked, the badges bot will post a message on the current channel letting everyone know that the user has been awarded a badge.
- If a subscription for this badge type is set, the badges bot will post a message on all subscribed channels letting everyone know that the user has been awarded a badge.

![Screenshot from 2022-03-16 12-28-45](https://user-images.githubusercontent.com/1933730/158580318-592bb139-6c43-48f0-99c3-79d868aa8024.png)

> If you try to award a not multiple badge to any user that already has it, nothing will happen (no messages will be created).

Reviewer: ```suggestion
If you try to award a badge that can't be awarded more than once to a single recipient, the badge won't be granted..
```

### Subscriptions
In order to create a subscription, you must be a badges admin.

> There are two ways to open the subscription creation dialog: `/badges subscription create` and the "Add badge subscription" menu from the channel menu.

Reviewer: See previous note about potential ways to introduce a step that has more than one way through to an end goal

Also - I'm a bit unclear about subscriptions and how they relate to badges. I feel that subscriptions may be a new concept at this point and may require a bit of explanation or context.
